replay_set.go

```diff
@@ -0,0 +1,60 @@
+package sphinx
+
+// ReplaySet is a data structure used to efficiently record the occurrence of
+// replays, identified by sequence number, when processing a Batch. Its primary
+// functionality includes set construction, membership queries, and merging of
+// replay sets.
+type ReplaySet struct {
+	replays map[uint16]struct{}
+}
+
+// NewReplaySet initializes an empty replay set.
+func NewReplaySet() *ReplaySet {
+	return &ReplaySet{
+		replays: make(map[uint16]struct{}),
+	}
+}
+
+// NewReplaySetFromBytes reconstructs a replay set given a byte slice. The byte
+// slice is assumed to be even in length, otherwise resulting in an
+// out-of-bounds panic.
+func NewReplaySetFromBytes(bs []byte) *ReplaySet {
+	rs := NewReplaySet()
+	for i := 0; i < len(bs); i += 2 {
+		seqNum := (uint16(bs[i]) << 8) | uint16(bs[i+1])
```

Comment:
What is the `bs` value meant to be?

Reply:
If it's a slice of 2 byte ints for each seqno to add, then this could be made a _bit_ more readable by using an `io.Reader`, and `encoding.Read`.

Reply:
Rewrote with `encoding.Read/Write`, much cleaner :)